### ISSUE TITLE
onUserSoundSample bug at 127 seconds after start

### ISSUE THREAD

**Issue by djm1176** [OneLoneCoder/videos#69]
On Windows 10 x64, I've been getting a bug where the frequency passed from `onUserSoundSample` is half the frequency I expect.
In order to reproduce this, the following should create a constant, 440hz sine wave for the first 127 seconds from the start. Afterwards, the pitch should be halved, and a 220hz sine wave is played:

```
virtual float onUserSoundSample(int nChannel, float fGlobalTime, float fTimeStep) {
    return sinf(440.0f * 2.0f * 3.14159f * fGlobalTime);
}
```

**Comment by djm1176** [comment]
I've noticed that `m_fGlobalTime` updates slower after 128 seconds as well. The issue is somewhere within the audio thread.

**Comment by OneLoneCoder** [comment]
Yes! This is a good one! It's a floating point number limitation. Either quick fix by moving to doubles, or you'll need to move to an integer/float hybrid for long term time accumulation.
